WalletWasabi/KeyManagement/KeyManager.cs

```diff
@@ -405,5 +454,72 @@ public bool AssertLockedInternalKeysIndexed(int howMany = 14)
 
 			return generated;
 		}
+
+		#region BlockchainState
+
+		public Height GetBestHeight()
+		{
+			lock (BlockchainStateLock)
+			{
+				return BlockchainState.BestHeight;
+			}
+		}
+
+		public IEnumerable<BlockState> GetTransactionIndex()
+		{
+			lock (BlockchainStateLock)
+			{
+				return BlockchainState.BlockStates.OrderBy(x => x).ToList();
```

Comment:
`BlockStates` are always ordered so, I think the `OrderBy` is redundant here. Also, I am not sure the `.ToList()` is necessary because the method returns an `IEnumerable` and `BlockStates` is already and `IEnumerable`.

Reply:
`ToList` is necessary because of the lock. I'll go through the orderbies however and make sure we add elements in order from the KeyManager inside so we don't have to do orderby here.